platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java

```diff
@@ -59,22 +60,33 @@ public int compare(V first, V second) {
 
 			if (isArgumentsCastableTo(first, second, IDisplayable.class)) {
 				var comparator = new CollatingDisplayStringFunctionBasedComparator<>(displayStringFunctionSupplier.get());
-				result = Comparator.nullsFirst(comparator).compare(first, second);
+				result = compareNullsFirst(comparator, first, second);
 			}
 
 			if (result == 0) {
 				if (isArgumentsCastableTo(first, second, IBasicItem.class)) {
 					var comparator = BasicItemComparator.get();
-					result = Comparator.nullsFirst(comparator).compare((IBasicItem) first, (IBasicItem) second);
+					result = compareNullsFirst(comparator, (IBasicItem) first, (IBasicItem) second);
 				}
 			}
 
 			return result;
 		}
 
 		else {
+			if (isArgumentsCastableTo(first, second, Integer.class)) {
+				return compareNullsFirst(Integer::compareTo, (Integer) first, (Integer) second);
+			} else if (isArgumentsCastableTo(first, second, Float.class)) {
+				return compareNullsFirst(Float::compareTo, (Float) first, (Float) second);
+			} else if (isArgumentsCastableTo(first, second, Double.class)) {
+				return compareNullsFirst(Double::compareTo, (Double) first, (Double) second);
+			} else if (isArgumentsCastableTo(first, second, BigDecimal.class)) {
+				return compareNullsFirst(BigDecimal::compareTo, (BigDecimal) first, (BigDecimal) second);
+			} else if (isArgumentsCastableTo(first, second, Long.class)) {
+				return compareNullsFirst(Long::compareTo, (Long) first, (Long) second);
+			}
```

Comment:
I know this is ugly, but every implementation of `Number` has a different `compareTo`, therefore I dont really know how to do this better than simple `else if` blocks